src/test/java/org/tests/transaction/TestExecuteComplete.java

```diff
@@ -11,75 +11,89 @@
 import io.ebean.annotation.Transactional;
 import io.ebeaninternal.api.SpiTransaction;
 import io.ebeaninternal.server.transaction.DefaultTransactionThreadLocal;
+
+import org.junit.After;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TestName;
 import org.tests.model.basic.Customer;
 import org.tests.model.basic.Order;
 
 import static org.assertj.core.api.StrictAssertions.assertThat;
+import static org.assertj.core.api.StrictAssertions.assertThatThrownBy;
 
 public class TestExecuteComplete extends BaseTestCase {
 
+  @Rule public TestName name = new TestName();
+
+  @After
+  public void checkForLeak() {
+    if (DefaultTransactionThreadLocal.clear()) {
+      System.out.println(name.getMethodName() + " had stuck transactionmap.");
+    } else {
+      System.out.println(name.getMethodName() + " cleared everything.");
```

Comment:
Currently, all tests are leaking a threadlocal object:
Transactionmap, key='h2', value = State(where transaction is null)

The next step I want to do, is to change the `TransactionMap.Map<String, State>` to `TransactionMap.Map<String, SpiTransaction>`